### DISCUSSION TITLE
Client developers comment here too please

### DISCUSSION THREAD

**Discussion by mattl** [foocorp/hacienda#91] [Community support for clients]
@jcorporation  Going to add you here.

Please make sure we have your information on https://github.com/foocorp/hacienda/wiki/Clients

**Comment by jcorporation** [comment]
Thanks @mattl

**Comment by FoxxMD** [comment]
Adding myself. I'd like to add libre.fm as a scrobble client for my application [multi-scrobbler](https://foxxmd.github.io/multi-scrobbler/) ([github](https://github.com/foxxmd/multi-scrobbler)) and am willing to write or help develop the typescript client for libre.fm to do this.

**Reply by arkag** [reply]
Did this ever get added?

**Reply by FoxxMD** [reply]
@arkag as far as I know there has been no progress on API development or documentation since I last asked about it here, about a year ago https://github.com/mattl/hacienda/discussions/96#discussioncomment-12657106

Multi-scrobbler is still actively developed, I added two new clients this month!

I would still love to implement a client for hacienda but there is nothing I can do until there is a usable API 🤷‍♀️

**Reply by mattl** [reply]
What's missing from the Last.fm/Audioscrobbler API that's already implemented?

**Reply by mattl** [reply]
I have started working on this: https://developer.libre.fm @FoxxMD

**Reply by FoxxMD** [reply]
@mattl thank you for the documentation effort.

So far I have [managed auth and basic api validation](https://github.com/FoxxMD/multi-scrobbler/issues/401) for `user.getInfo`, but `user.getRecentTracks` always returns `"{\"error\":{\"#text\":\"Invalid resource specified\",\"code\":\"7\"}}"` as the api response.

EDIT: I should specify im trying to use the last.fm 2.0 compatibility at `libre.fm/2.0/`

**Reply by FoxxMD** [reply]
Seems everything is okay-*ish*. My account had zero scrobbles. After scrobbling anything, `getRecentTracks` is now returning a proper response. Would be nice if it returned an empty `tracks` array instead of throwing an opaque error.

**Comment by GeopJr** [comment]
Adding myself for [Turntable](https://turntable.geopjr.dev/) - [Flathub](https://flathub.org/apps/dev.geopjr.Turntable) | [Source](https://codeberg.org/GeopJr/Turntable)
https://github.com/foocorp/wiki.libre.fm/pull/18
